src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -289,78 +300,79 @@ TCPTransactionId RTCPMessageManager::sendConnectionRequest(TCPChannelResource *p
     //logError(DEBUG, "Sending Connection Request with locator: " << IPLocator::to_string(request.transportLocator()));
     TCPTransactionId id = getTransactionId();
-    bool success = sendData(p_channel_resource, BIND_CONNECTION_REQUEST, id, &payload);
+    bool success = sendData(channel, BIND_CONNECTION_REQUEST, id, &payload);
     if (!success)
     {
         logError(RTCP, "Failed sending Connection Request");
     }
-    p_channel_resource->change_status(TCPChannelResource::eConnectionStatus::eWaitingForBindResponse);
+    channel->change_status(TCPChannelResource::eConnectionStatus::eWaitingForBindResponse);
     return id;
 }
 
 TCPTransactionId RTCPMessageManager::sendOpenLogicalPortRequest(
-        TCPChannelResource *p_channel_resource,
+        TCPChannelResource* channel,
         uint16_t port)
 {
     OpenLogicalPortRequest_t request;
     request.logicalPort(port);
-    return sendOpenLogicalPortRequest(p_channel_resource, request);
+    return sendOpenLogicalPortRequest(channel, request);
 }
 
 TCPTransactionId RTCPMessageManager::sendOpenLogicalPortRequest(
-        TCPChannelResource *p_channel_resource,
+        TCPChannelResource* channel,
         OpenLogicalPortRequest_t &request)
 {
     SerializedPayload_t payload(static_cast<uint32_t>(OpenLogicalPortRequest_t::getBufferCdrSerializedSize(request)));
     request.serialize(&payload);
     logInfo(RTCP_MSG, "Send [OPEN_LOGICAL_PORT_REQUEST] LogicalPort: " << request.logicalPort());
     TCPTransactionId id = getTransactionId();
-    sendData(p_channel_resource, OPEN_LOGICAL_PORT_REQUEST, id, &payload);
+    sendData(channel, OPEN_LOGICAL_PORT_REQUEST, id, &payload);
     return id;
 }
 
 TCPTransactionId RTCPMessageManager::sendCheckLogicalPortsRequest(
-        TCPChannelResource *p_channel_resource,
+        TCPChannelResource* channel,
         std::vector<uint16_t> &ports)
 {
     CheckLogicalPortsRequest_t request;
     request.logicalPortsRange(ports);
-    return sendCheckLogicalPortsRequest(p_channel_resource, request);
+    return sendCheckLogicalPortsRequest(channel, request);
 }
 
 TCPTransactionId RTCPMessageManager::sendCheckLogicalPortsRequest(
-        TCPChannelResource *p_channel_resource,
+        TCPChannelResource* channel,
         CheckLogicalPortsRequest_t &request)
 {
     SerializedPayload_t payload(static_cast<uint32_t>(CheckLogicalPortsRequest_t::getBufferCdrSerializedSize(request)));
     request.serialize(&payload);
     logInfo(RTCP_MSG, "Send [CHECK_LOGICAL_PORT_REQUEST]");
     TCPTransactionId id = getTransactionId();
-    sendData(p_channel_resource, CHECK_LOGICAL_PORT_REQUEST, id, &payload);
+    sendData(channel, CHECK_LOGICAL_PORT_REQUEST, id, &payload);
     return id;
 }
 
 TCPTransactionId RTCPMessageManager::sendKeepAliveRequest(
-        TCPChannelResource *p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.